src/riak_kv_vnode.erl

```diff
@@ -1046,8 +1046,11 @@ handle_handoff_command(Req=?KV_PUT_REQ{}, Sender, State) ->
     end;
 
 handle_handoff_command(?KV_W1C_PUT_REQ{}=Request, Sender, State) ->
-    {noreply, NewState} = handle_command(Request, Sender, State),
-    {forward, NewState};
+    NewState0 = case handle_command(Request, Sender, State) of
+        {noreply, NewState} -> NewState;
+        {reply, _Reply, NewState} -> NewState
```

Comment:
After talking with Jon, in the {reply, ...} case we should send the reply ourselves with riak_core_vnode:reply/? and then continue to forward the original put request to the new vnode. Otherwise, the current owner will not respond in the non-async case.